src/view.c

```diff
@@ -2015,7 +2015,13 @@ void
 mappable_disconnect(struct mappable *mappable)
 {
 	assert(mappable);
-	assert(mappable->connected);
+	/*
+	 * pygame-zero gets us here without a mappable_connect() first so we
+	 * have to handle this gracefully. See issue #1466
+	 */
+	if (!mappable->connected) {
+		return;
```

Comment:
I'd suggest we still print a warning here ("received dissociate event but no surface was associated" or similar). If it shows up in the log when debugging some other issue, it could provide clues that a client and/or wlroots is doing something weird.